src/reducers/user.js

```diff
@@ -0,0 +1,106 @@
+import { handleActions } from 'redux-actions';
+import update from 'immutability-helper';
+import * as constants from '../action/constants';
+
+export const initialState = {
+  userLogin: {
+    data: [],
+    error: [],
+    isSuccess: false,
+    isError: false,
+    isLoading: true,
+  },
+  deviceData: {
+    data: [],
+    error: [],
+    isSuccess: false,
+    isError: false,
+  },
+};
+
+const userLoginSuccess = (state, action) => update(state, {
+  userLogin: {
+    data: { $set: action.payload.data },
+    isLoading: { $set: false },
+    isSuccess: { $set: true },
+    isError: { $set: false },
+    errors: { $set: [] },
+  },
+});
+
+const userLoginFailed = (state, action) => update(state, {
+  userLogin: {
+    data: { $set: [] },
+    isLoading: { $set: false },
+    isSuccess: { $set: false },
+    error: { $set: action.payload.data },
+    isError: { $set: true },
+  },
+});
+
+
+const deviceDataSuccess = (state, action) => update(state, {
+  deviceData: {
+    data: { $set: action.payload.data },
+    isLoading: { $set: false },
+    isSuccess: { $set: true },
+    isError: { $set: false },
+    errors: { $set: [] },
+  },
+});
+
+const deviceDataFailed = (state, action) => update(state, {
+  deviceData: {
+    data: { $set: [] },
+    isLoading: { $set: false },
+    isSuccess: { $set: false },
+    errors: { $set: action.payload.data },
+    isError: { $set: true },
+  },
+});
+
+const userLogoutSuccess = (state, action) => update(state, {
+  login: {
+    data: { $set: action.payload.data },
+    isLoading: { $set: false },
```

Comment:
this is not correct. 

need to create helpers using immutable using the $extend function